node/src/p2p/callbacks/p2p_callbacks_reducer.rs

```diff
@@ -290,6 +296,67 @@ impl crate::State {
                     best_tip: best_tip.clone(),
                 });
             }
+            P2pCallbacksAction::P2pPubsubValidateMessage { message_id } => {
+                let Some(message_content) = state.p2p.ready().and_then(|p2p| {
+                    p2p.network
+                        .scheduler
+                        .broadcast_state
+                        .mcache
+                        .get_message(message_id)
+                }) else {
+                    return;
+                };
+
+                let pre_validation_result = match message_content {
+                    GossipNetMessageV2::NewState(new_best_tip) => {
+                        match BlockWithHash::try_new(new_best_tip.clone()) {
+                            Ok(block) => {
+                                let allow_block_too_late = allow_block_too_late(state, &block);
+                                match state.prevalidate_block(&block, allow_block_too_late) {
+                                    Ok(()) => PreValidationResult::Continue,
+                                    Err(BlockPrevalidationError::ReceivedTooEarly { .. }) => {
+                                        PreValidationResult::Ignore
+                                    }
+                                    Err(_) => PreValidationResult::Reject,
+                                }
+                            }
+                            Err(_) => {
+                                log::error!(time; "P2pCallbacksAction::P2pPubsubValidateMessage: Invalid bigint in block");
+                                PreValidationResult::Reject
+                            }
+                        }
+                    }
+                    _ => {
+                        // TODO: add pre validation for Snark pool and Transaction pool diffs
+                        PreValidationResult::Continue
+                    }
+                };
+
+                match pre_validation_result {
+                    PreValidationResult::Continue => {
+                        dispatcher.push(P2pNetworkPubsubAction::ValidateIncomingMessage {
+                            message_id: *message_id,
+                        });
+                    }
+                    PreValidationResult::Reject => {
+                        dispatcher.push(P2pNetworkPubsubAction::RejectMessage {
+                            message_id: Some(p2p::BroadcastMessageId::MessageId {
+                                message_id: *message_id,
+                            }),
+                            peer_id: None,
+                            reason: "Pre validation failed".to_owned(),
```

Comment:
I would like the reason to be included here. It is contained in the result of `prevalidate_block` but discarded.